zcash_client_sqlite/src/wallet.rs

```diff
@@ -202,9 +202,13 @@ impl Account {
         &self,
         request: UnifiedAddressRequest,
     ) -> Result<(UnifiedAddress, DiversifierIndex), AddressGenerationError> {
+        self.uivk().default_address(request)
```

Comment:
This refactor of `Account::default_address` OTOH is fine; its previous internals were somewhat duplicative and this ends up cleaner.